tests/test_build_linkcheck.py

```diff
@@ -625,3 +625,14 @@ def test_get_after_head_raises_connection_error(app):
         "uri": "http://localhost:7777/",
         "info": "",
     }
+
+
+@pytest.mark.sphinx('linkcheck', testroot='linkcheck-documents_exclude', freshenv=True)
+def test_linkcheck_exclude_documents(app):
+    with http_server(make_redirect_handler(support_head=False)):
+        app.build()
```

Comment:
No HTTP request will be made.
```suggestion
    app.build()
```